core/src/main/java/com/statful/client/core/api/MetricsSenderAPI.java

```diff
@@ -107,38 +204,23 @@ public SenderAPI aggregations(final Aggregation... aggregations) {
     }
 
     @Override
-    public SenderAPI aggregations(final Aggregations aggregations) {
+    public MetricsSenderAPI aggregations(final Aggregations aggregations) {
```

Comment:
Probably this code will not evolve that much - we are never really sure :) - but I think coupling the caller with the specific implementation instead of the interface will probably make those changes a lot more difficult.